keras/backend/theano_backend.py

```diff
@@ -1877,6 +1877,10 @@ def conv2d_transpose(x, kernel, output_shape, strides=(1, 1),
     else:
         # Will only work if `kernel` is a shared variable.
         kernel_shape = kernel.eval().shape
+
+    if padding == 'same' and kernel_shape[0] % 2 == 0:
+        raise ValueError('Even kernel sizes are only supported with Tensorflow.')
```

Comment:
Please provide a more detailed error message, including a suggestion of what the user should do to fix the issue. This message would not be immediately understandable by users.